### ISSUE TITLE
Pitfall 2 not working

### ISSUE THREAD

**Issue by sa666666** [stella-emu/stella#661]
Pitfall 2 stopped working in the latest release.  Original report at https://atariage.com/forums/topic/307760-stella-62-released/page/2/?tab=comments#comment-4560509.

**Comment by thrust26** [comment]
@sa666666 What exactly was the problem? I suppose it was related to the last minute rounding issue?

**Comment by sa666666** [comment]
Well, it was related to a last-minute issue, but not the rounding one.  It was my fault :disappointed: I added code to pass in the desired bankswitch size.  However, we ran into the issue of 3E and 3E+ having different sizing requirements.  So I added a delegating c'tor to 3E.  However, the real issue was with inheritance, and a base class and derived class having different desired sizes.  3E/3E+ was only one instance.  I missed F8/DPC, and upon further inspection FA/FA2 as well.

My new solution is to pass in the desired size for _all_ cart classes that eventually inherit from `CartEnhanced`.  I made it a default parameter, so that the rest of the code isn't affected.  But it also solves this problem for good, since each derived class is now responsible for passing in its desired size.

So long story short; F8 defaulted to 8_KB size, but DPC inherited from it and needed 10_KB.  But the 10_KB wasn't being passed to the base class, so the image was essentially truncated.  In fact, it's a wonder it worked at all, since there was a pointer to memory we didn't even have??
